tests/components/cloud/test_google_config.py

```diff
@@ -865,3 +865,43 @@ async def test_google_config_get_agent_user_id(
         == config.agent_user_id
     )
     assert config.get_agent_user_id_from_webhook("other_id") != config.agent_user_id
+
+
+async def test_google_config_get_agent_users(
+    hass: HomeAssistant, mock_cloud_login, cloud_prefs
+) -> None:
+    """Test overridden async_get_agent_users method."""
+    username_mock = PropertyMock(return_value="blah")
+
+    # We should not call Cloud.username when not logged in
+    cloud_prefs._prefs["google_connected"] = True
```

Comment:
We have the `set_cloud_prefs` fixture we could use if we also use the `cloud` fixture.